openshift_tools/monitoring/hawk_client.py

```diff
@@ -0,0 +1,113 @@
+from openshift_tools.monitoring.metricmanager import UniqueMetric, MetricManager
+from openshift_tools.monitoring.hawk_common import HawkConnection
+
+# The hawkular client
+from hawkular.metrics import HawkularMetricsClient, MetricType, Availability
+
+#This class implements rest calls. We only have one rest call implemented
+# add-metric.  More could be added here
+#pylint: disable=too-few-public-methods
+class HawkClient(object):
+    """
+    wrappers class around hawkular client python so use can use it with UniqueMetric
+    """
+
+    def __init__(self, hawk_connection, headers=None):
+        # pylint doesn't know where RestAPI is
+        #pylint: disable=undefined-variable
+        self.hawk_conn = hawk_connection
+        self.client = None
+
+        # Do not create a client if inactive
+        if not self.hawk_conn.active:
+            return
+
+        self.client = HawkularMetricsClient(host=self.hawk_conn.host,
+                                            port=self.hawk_conn.port,
+                                            scheme=self.hawk_conn.scheme,
+                                            username=self.hawk_conn.username,
+                                            password=self.hawk_conn.password,
+                                            context=self.hawk_conn.context,
+                                            tenant_id=self.hawk_conn.tenant_id,
+                                           )
+
+    def push_heartbeat(self, metric):
+        """
+        Push a list of heartbeat metric-endpoints via hawkular client
+        """
+        # We assume the value of a heartbeat is a dict
+        heartbeat = metric.value
+        # Hawkular metrics heartbeat value is "up"
+        value = Availability.Up
+        # Hawkular metrics use milliseconds
+        clock = metric.clock * 1000
+        # Use MetricType.Availability for heartbeat data
+        metric_type = MetricType.Availability
+
+        for template in heartbeat.get('templates'):
+            # Add the group and host to the key
+            key = '{0}/{1}/{2}'.format('ops/heartbeat', 'template', template)
+            self.client.push(metric_type, key, value, clock)
+
+        for hostgroup in heartbeat.get('hostgroups'):
+            # Add the group and host to the key
+            key = '{0}/{1}/{2}'.format('ops/heartbeat', 'hostgroup', hostgroup)
+            self.client.push(metric_type, key, value, clock)
+
+    def add_metric(self, unique_metric_list):
+        """
+        Add a list of UniqueMetrics (unique_metric_list) via hawkular client
+        """
+        # Do not run if inactive
+        if not self.client:
+            return
+
+        metric_list = [m for m in unique_metric_list if m.key != 'heartbeat']
+        heartbeat_list = [m for m in unique_metric_list if m.key == 'heartbeat']
+
+        for metric in metric_list:
+            # Hawkular metrics support only numeric values
+            value = float(metric.value)
```

Comment:
It need to be numeric float or Int 
I wanted to raise an exception if not numeric, but it's safe to remove it here if all data is numeric ( e.g. ints )
it will be converted to a string when sent using the Rest API :-)